core/src/main/java/org/apache/struts2/interceptor/csp/CspInterceptor.java

```diff
@@ -124,4 +153,11 @@ public void setPrependServletContext(boolean prependServletContext) {
         this.prependServletContext = prependServletContext;
     }
 
-}
+    /**
+     * Sets the class name of the default {@link CspSettings} implementation to use when the action does not
+     * set its own values. If not set, the default is {@link DefaultCspSettings}.
+     */
+    public void setDefaultCspSettingsClassName(String defaultCspSettingsClassName) {
+        this.defaultCspSettingsClassName = defaultCspSettingsClassName;
+    }
```

Comment:
You can use Struts inject mechanism instead of using raw class and creating the instance by yourself. It's all about defining a `<bean name="customCspSettings" class="..."/>` and then annotating the setter with `@Inject("customCspSettings")`.

I assume you never played with Struts @Inject, so let's leave it as is and I will change that in the next PR.